core/src/replay_stage.rs

```diff
@@ -1919,16 +1919,16 @@ impl ReplayStage {
             Some((_stake, vote_account)) => vote_account,
         };
         let vote_state = vote_account.vote_state();
-        let vote_state = match vote_state.as_ref() {
-            Err(_) => {
-                warn!(
-                    "Vote account {} is unreadable.  Unable to vote",
-                    vote_account_pubkey,
-                );
-                return None;
-            }
-            Ok(vote_state) => vote_state,
-        };
+        if vote_state.is_err() {
+            drop(vote_state);
+            warn!(
+                "Vote account {} is unreadable.  Unable to vote",
+                vote_account_pubkey,
+            );
+            return None;
+        }
+        // SAFETY: The `.unwrap()` is safe here because we just checked if vote_state is Err
+        let vote_state = vote_state.as_ref().unwrap();
```

Comment:
This doesn't drop the read lock so other IO below will also be done while holding the lock. Can we maybe pull out the info we need and then drop the read lock explicitly as early as possible?

Reply:
Correct. Yeah I wasn't sure how long the lock _needed_ to be held here so I decided that my first-pass would be to preserve the current behavior.

My preference would be to change behavior in a subsequent PR. Would that be ok for you, or would you rather see those changes in this PR?